### PR TITLE
🐛 Fix new labeler config steps

### PR DESCRIPTION
**Pull request by foysalit**
This PR ensures that initial labeler set up steps can continue on without getting blocked by attempts to fetch server config.

### CODE REVIEW

File: components/shell/ConfigurationFlow.tsx [hunk 1]
```diff
@@ -241,6 +243,19 @@ function IdentityConfigurationFlow({
             placeholder="Confirmation Code"
             className="block w-full mt-2"
             value={token}
+            autoFocus
```
Comment (foysalit) [inline components/shell/ConfigurationFlow.tsx:246]: These are unrelated UX improvements.

File: app/actions/ModActionPanel/QuickAction.tsx [hunk 2]
```diff
@@ -206,6 +207,7 @@ function Form(
   const deactivatedAt = getDeactivatedAt(
     repo ? { repo } : record ? { record } : {},
   )
+  const canManageChat = checkPermission('canManageChat')
```
Comment (foysalit) [inline app/actions/ModActionPanel/QuickAction.tsx:210]: Also unrelated but this hides the `show messaging data` block from UI for external labelers.

File: components/common/labels/List.tsx [hunk 3]
```diff
@@ -97,7 +97,7 @@ export const ModerationLabel = ({
   const labelVal = toLabelVal(label, recordAuthorDid)
   const isSelfLabeled = isSelfLabel(labelVal)
   const labelDefFromService =
-    labelerServiceDef?.policies.definitionById[label.val]
+    labelerServiceDef?.policies.definitionById?.[label.val]
```
Comment (foysalit) [inline components/common/labels/List.tsx:100]: This can crash a page if a plain label is added when the policy definition does not exist. So, this is more of a safety fallback.

File: components/shell/ConfigurationFlow.tsx [hunk 4]
```diff
@@ -164,7 +164,9 @@ export function ConfigurationFlow() {
         key={config.updatedAt}
         session={session}
         onComplete={(skip) => {
-          client.reconfigure({ skipRecord: skip })
+          client
+            .reconfigure({ skipRecord: skip })
+            .then(() => client.refetchServerConfig())
```
Comment (devinivy) [inline components/shell/ConfigurationFlow.tsx:169]: I wouldn't expect the server config to be updated from the labeler record configuration flow—is it possible we don't need it?

File: lib/client.ts [hunk 5]
```diff
@@ -79,7 +79,12 @@ class ClientManager extends EventTarget {
       password,
     })
     const config = await this._getConfig()
-    const serverConfig = await this._getServerConfig(agent, config.did)
+    // If the labeler needs further configuration, let's generate a mock serverConfig
+    // and let the user move forward with configuration steps
+    const needsConfiguration = config.needs.key || config.needs.service
+    const serverConfig = await (needsConfiguration
+      ? parseServerConfig({})
+      : this._getServerConfig(agent, config.did))
```
Comment (devinivy) [inline lib/client.ts:87]: I'm not sure I understand how this part works.  Ideally IMO the server config and the labeler config could be managed independently from each other, and we wouldn't need to avoid loading the real server config.  But maybe there's an overlap that I'm overlooking.

Comment (foysalit) [reply lib/client.ts:87]: requests to the server config seems to fail until the labeler config is available. the error was something like `record not found` or something along that line and since it doesn't make a lot of sense to fetch this until the labeler config is available, delaying the fetch makes sense to me.

File: components/shell/ConfigurationFlow.tsx (hunk 4 above)
Comment (foysalit) [reply components/shell/ConfigurationFlow.tsx:169]: it doesn't update but this is the issue i was seeing https://github.com/bluesky-social/ozone/pull/156#discussion_r1705843465 so after the configuration is complete, we are fetching the server config for real here.

File: lib/client.ts (hunk 5 above)
Comment (devinivy) [reply lib/client.ts:87]: (And if we're not sure whether we interpreted the "record not found" error properly, perhaps there's still an issue out there!)

Comment (devinivy) [reply lib/client.ts]: Ohhhhkay, I get what's going on now.  They are tangled together a little bit because we need to use service proxying to fetch the server config.  And server proxying is not going to work until the service is reflected in the labeler's DID document.  So we need to wait until `config.needs.service` is `false` before we can make authed requests to ozone.

I made some small tweaks in https://github.com/bluesky-social/ozone/pull/156/commits/fc7093cb460d98ba09518873a91d9788123fc919, let me know what you think!

**Review by devinivy — APPROVED** [review]
Sweet!  I tried this out, working well 👍
